### PR TITLE
 Refactor UI software design to MVC Pattern

### PR DESCRIPTION
**Pull request by okucheruk**
This PR refactors UI software design to MVC Pattern.

**Highlights:**
- Added `AuthenticationController` to take a control of `AuthenticationPage` view and its events.
- Added `SignupController` to take control of `SignupPage` view and its events.
- Project file structure refactored to follow Coupling Pattern.

### CODE REVIEW

**Comment by okucheruk** [comment]
@dkalinin, PTAL.

File: app/auth/AuthenticationController.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    this._view = new AuthenticationPage(parentElement, (event) => this.handlesSigninButton(event));
+  }
+
+  /**
+   * Handles 'click' event type of {@link Button} UI component 'sign in'.
+   *
+   * @param {Event} event The event which takes place in the DOM.
+   * @private
+   */
+  handlesSigninButton(event) {
```
Comment (dkalinin) [inline app/auth/AuthenticationController.js]: handleSignIn

File: app/auth/AuthenticationController.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    event.stopPropagation();
+    event.preventDefault();
+
+    console.log('Nice');
+
+    const context = this._view.getSigninContext();
```
Comment (dkalinin) [inline app/auth/AuthenticationController.js]: Let's get it from event.

File: app/signup/SignupPage.js [hunk 3]
```diff
@@ -117,8 +131,8 @@ export default class SignUpPage extends DivContainer {
     new Button(parentElement)
         .withClassName('mui-btn')
         .withTextContent('Sign Up')
-        .subscribeToEvent('click',
-            (event) => this._signupButtonClickHandler(event));
+        .subscribeToEvent('click', (event) => this._controlCallback(event));
+    // .subscribeToEvent('click', (event) => this._signupButtonClickHandler(event));
```
Comment (dkalinin) [inline app/signup/SignupPage.js]: Please remove.

File: app/auth/AuthenticationController.js (hunk 1 above)
Comment (okucheruk) [reply app/auth/AuthenticationController.js]: Refactored.

File: app/signup/SignupPage.js (hunk 3 above)
Comment (okucheruk) [reply app/signup/SignupPage.js]: Removed.

File: app/auth/AuthenticationController.js (hunk 2 above)
Comment (okucheruk) [reply app/auth/AuthenticationController.js]: Refactored.
`new AuthenticationPage(parentElement, (event, context) => this.handlesSignIn(event, context));`

**Comment by okucheruk** [comment]
@dkalinin, fixed, PTAL.

File: app/auth/AuthenticationController.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   */
+  handlesSignIn(event, context) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    new HttpRequestService()
+        .post(context)
+        .then((response) => {
+          console.log(response);
+          SpaRouter.redirectTo('/FileBrowser');
```
Comment (dkalinin) [inline app/auth/AuthenticationController.js]: /filebrowser

Comment (okucheruk) [reply app/auth/AuthenticationController.js]: Changed to lowercase.

**Pull request merged**